### ISSUE TITLE
Barely functional with Calckey 14.0.0-dev19-hk25 (not sure if supported?)

### ISSUE THREAD

**Issue by transponderings** [feditext/feditext#143]
**To Reproduce**
Log in to an instance of Calckey 14.0.0-dev19-hk25. In my case this was blahaj.zone.

That’s enough to produce the JSON report below.

(Switching timelines produces the same problem – no timeline is visible._

**Expected behavior**
The timelines should have been visible, or the app should have gracefully declined to log in.

**Screenshots**
This screenshot shows how my blahaj.zone profile appears:
![image](https://github.com/feditext/feditext/assets/88622893/f61031cb-1d59-4411-ad09-891901d7e92a)

**(please complete the following information):**
 - Device: iPhone XS
 - OS: iOS 16.5.1
 - App Version: 1.7.2 (67)

**Additional context**
JSON report:

```json
{
  "alertLocation" : {
    "file" : "ViewModels/CollectionItemsViewModel.swift",
    "function" : "request(maxId:minId:search:)",
    "line" : 370
  },
  "app" : {
    "build" : "67",
    "bundle" : "com.gotgoat.metatext",
    "version" : "1.7.2"
  },
  "error" : {
    "error" : {
      "valueNotFound" : {
        "context" : {
          "codingPath" : [
            "_JSONKey(stringValue: \"Index 0\", intValue: 0)",
            "CodingKeys(stringValue: \"account\", intValue: nil)",
            "CodingKeys(stringValue: \"avatarStatic\", intValue: nil)"
          ],
          "debugDescription" : "Expected String but found null value instead."
        },
        "type" : "Swift.String"
      }
    },
    "method" : "GET",
    "requestLocation" : {
      "file" : "ServiceLayer/TimelineService.swift",
      "function" : "request(maxId:minId:search:)",
      "line" : 93
    },
    "url" : "https://blahaj.zone/api/v1/timelines/home"
  },
  "message" : "GET https://blahaj.zone/api/v1/timelines/home\n\nvalueNotFound(type: \"Swift.String\", context: HTTP.AnnotatedDecodingError.EncodableContext(codingPath: [\"_JSONKey(stringValue: \\\"Index 0\\\", intValue: 0)\", \"CodingKeys(stringValue: \\\"account\\\", intValue: nil)\", \"CodingKeys(stringValue: \\\"avatarStatic\\\", intValue: nil)\"], debugDescription: \"Expected String but found null value instead.\", underlyingError: nil))",
  "system" : {
    "idiom" : "phone",
    "model" : "iPhone",
    "name" : "iOS",
    "version" : "16.5.1"
  },
  "title" : "AnnotatedDecodingError",
  "type" : "AnnotatedDecodingError"
}
```

**Comment by VyrCossont** [comment]
This looks like a bug in the Mastodon compatibility API for Hajkey: [`Account.avatar_static`](https://docs.joinmastodon.org/entities/Account/#avatar_static) is not optional. It's something we can compensate for, but it should really be logged as a bug against Hajkey as well, since it's a break with the Mastodon API.

**Comment by transponderings** [comment]
I hadn’t even realised that Hajkey or any of the Misskey/Calckey derivatives had a Mastodon-compatible API. This is the first Mastodon app I’ve seen that gets anywhere near working with them. I have a Firefish account that seems to work ok.

(I don’t know how to, and don’t currently have the spoons to, log bugs against Hajkey too, I’m afraid.)

**Comment by VyrCossont** [comment]
No worries!

Feditext doesn't currently and will likely never support Misskey's native API, so we use the Mastodon compatibility API on Misskey derivatives that have one.

I'm going to work around this bug in our next build, but I haven't read the code for or done a compatibility test against any Misskey derivative other than Firefish, so I'm not sure what's supposed to work. I _was_ able to log into a blahaj.zone account after patching this, so it should get blahaj.zone users unblocked at least to the point of being able to read their timelines.

**Comment by josharky** [comment]
I've also opened issue at https://codeberg.org/supakaity/hajkey/issues/13 referencing this issue which is same as mine.

**Comment by VyrCossont** [comment]
@josharky this one is closely related to the one you reported at https://blahaj.zone/notes/9hwkfahq16

I'm looking at data from https://blahaj.zone/api/v2/suggestions and it looks like the output is often missing image assets, and every single account URL is wrong (they're the exact same string same as the `acct` account name). On top of that, some account bios aren't rendering correctly, or are being replaced and then overwritten by partial versions of the same account data.

tl;dr: this is going to turn into quite a few Iceshrimp bugs, a few of which Feditext can safely work around by slightly relaxing compatibility with the Mastodon API, most of which need to be fixed server-side.

**Comment by bdube** [comment]
My apologies. This was closed by mistake in a branching accident.

**Comment by VyrCossont** [comment]
One specific action item for Feditext: I think following up on 7ae4eb722288d60eb5931c0d0a37320ad32b8fd2 to make all account image URL fields optional might be a good idea, since we can already display an account that doesn't have a header or avatar, and since leaving out image assets that don't exist is apparently a common break from strict Mastodon compatibility. When that's done, I'll close this Feditext bug.

The issues with the suggestions API have already been fixed in Firefish, and probably in Iceshrimp, so I'm not going to bother opening a Hajkey bug for stuff like this until Hajkey rebases on Iceshrimp.

**Comment by VyrCossont** [comment]
I'm going to close this since it's not linked to any known concrete issues. Will reopen or open a related issue if that changes.
